client/ayon_houdini/api/workfile_template_builder.py

```diff
@@ -0,0 +1,292 @@
+        keep_placeholder = profile.get("keep_placeholder")
+        create_first_version = profile.get("create_first_version")
+
+        # backward compatibility, since default is True
+        if keep_placeholder is None:
+            keep_placeholder = True
+
+        if not path:
+            raise TemplateLoadFailed((
+                "Template path is not set.\n"
+                "Path need to be set in {}\\Template Workfile Build "
+                "Settings\\Profiles"
+            ).format(host_name.title()))
+
+        # Try fill path with environments and anatomy roots
+        anatomy = Anatomy(project_name)
+        fill_data = {
+            key: value
+            for key, value in os.environ.items()
+        }
+
+        fill_data["root"] = anatomy.roots
+        fill_data["project"] = {
+            "name": project_name,
+            "code": anatomy.project_code,
+        }
+
+        result = StringTemplate.format_template(path, fill_data)
+        if result.solved:
+            path = result.normalized()
+
+        # I copied the whole thing because I wanted to add some
+        # Houdini specific code here
+        path = hou.text.expandString(path)
+
+        if path and os.path.exists(path):
+            self.log.info("Found template at: '{}'".format(path))
+            return {
+                "path": path,
+                "keep_placeholder": keep_placeholder,
+                "create_first_version": create_first_version
+            }
+
+        solved_path = None
+        while True:
+            try:
+                solved_path = anatomy.path_remapper(path)
+            except KeyError as missing_key:
+                raise KeyError(
+                    "Could not solve key '{}' in template path '{}'".format(
+                        missing_key, path))
+
+            if solved_path is None:
+                solved_path = path
+            if solved_path == path:
+                break
+            path = solved_path
+
+        solved_path = os.path.normpath(solved_path)
+        if not os.path.exists(solved_path):
+            raise TemplateNotFound(
+                "Template found in AYON settings for task '{}' with host "
+                "'{}' does not exists. (Not found : {})".format(
+                    task_name, host_name, solved_path))
+
+        self.log.info("Found template at: '{}'".format(solved_path))
+
+        return {
+            "path": solved_path,
+            "keep_placeholder": keep_placeholder,
+            "create_first_version": create_first_version
+        }
+
+    def import_template(self, path):
+        """Import template into current scene.
+        Block if a template is already loaded.
+
+        Args:
+            path (str): A path to current template (usually given by
+            get_template_preset implementation)
+
+        Returns:
+            bool: Whether the template was successfully imported or not
+        """
+
+        # TODO Check if template is already imported
+
```

Comment:
```suggestion
        if self.host.get_current_workfile() == path:
                raise RuntimeError("The user's current filepath should not be imported template file")

```

Reply:
This is a nice idea!
I thought about checking the file contents.

Reply:
This won't work anymore if we resolve the issue that the current filepath even becomes the template, because it shouldn't ;)

Reply:
marking is as resolved. `import_template` function no longer opens the template file but merges its content into the current scene.